app/lib/event_store/event.rb

```diff
@@ -66,6 +66,35 @@ def load(str)
 
     scope :stale, -> { where.has { created_at <= TTL.ago } }
 
+    class EventRollbackError < StandardError
+      include Bugsnag::MetaData
+
+      def initialize(error:, event:)
+        event_details = event.to_h.as_json
+        self.bugsnag_meta_data = { event: event_details }
+
+        super "Error raised trying to roll back #{event}: #{self}, event details: #{event_details}"
+        set_backtrace error.backtrace
+      end
+    end
+
+    # Since Rails 6.1 the EventStore::Event records (among others) are added to the
+    # list of the objects, belonging to a current transaction, and #rolledback! is called on them
+    # if the transaction is rolled back on the DB.
+    # The problem is that this triggers the Event's `load` and `deserialize` (to restore the original state), and if
+    # the object referenced in the event via GlobalID does not exist, the deserialization of the event will fail.
+    # This causes some tests (that have explicit transactions and rollbacks) to fail, and can potentially cause issues
+    # in production as well.
```

Comment:
Reference to the PR/commit would be useful here so that one doesn't have to lookup the upstream change.